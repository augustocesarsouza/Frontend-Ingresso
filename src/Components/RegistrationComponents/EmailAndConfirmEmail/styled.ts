import styled from 'styled-components';

export const ContainerMainEmail = styled.div`
  display: flex;
  gap: 20px;
`

interface ContainerWarningProps {
  $container: string;
}

export const ContainerWarning = styled.div<ContainerWarningProps>`
  width: ${props => props.$container === "1" ? "55%" : "42%"};
`

interface ContainerColorWarningProps {
  $focusorblur: string;
  $emailvalid: string;
}

export const ContainerColorWarning = styled.div<ContainerColorWarningProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  border-width: 1px 1px 1px 5px;
  border-style: solid;
  border-color: #8080804d;
  border-color: ${props => props.$emailvalid === "false" ? "red" : props.$focusorblur === "true" ? "#2196f3" : "#8080804d"};
  margin-bottom: 6px;
  /* width: 55%; */
`

interface ContainerEmailSvgProps {
  $focus: string;
  $emailvalid: string;
}

export const ContainerEmailSvg = styled.div<ContainerEmailSvgProps>`
  /* position: absolute; */
  left: 0;
  top: 0;
  width: 45px;
  height: 22px;
  border-right: 1px solid rgb(221, 221, 221);

  svg {
    width: 100%;
    height: 100%;
    fill: ${props => props.$emailvalid === "false" ? "red" : props.$focus === "true" ? "#2196f3" : "#8080804d"};
  }
`

interface SpanWarningProps {
  $hasvalueinput: string;
}

export const SpanWarning = styled.span<SpanWarningProps>`
  position: absolute;
  left: 53px;
  top: ${props => props.$hasvalueinput === "true" ? "4px" : "13px"};
  font-size: ${props => props.$hasvalueinput === "true" ? "11px" : "13px"};
  color: #a4a2a2;
  pointer-events: none;
  transition: top 0.2s ease, font-size 0.2s ease;
`

interface InputProps {
  $hasvalueinputname: string;
}

export const Input = styled.input<InputProps>`
  border: none;
  width: 100%;
  height: 100%;
  outline: none;
  padding: ${props => props.$hasvalueinputname === "true" ? "0px" : "13px"};
  padding-top: ${props => props.$hasvalueinputname === "true" && "18px"};
  padding-left: ${props => props.$hasvalueinputname === "true" && "17px"};
  padding-bottom: ${props => props.$hasvalueinputname === "true" && "8px"};
`

interface InputConfirmEmailProps {
  $hasvalueinputconfirmemail: string;
}

export const InputConfirmEmail = styled.input<InputConfirmEmailProps>`
  border: none;
  width: 100%;
  height: 100%;
  outline: none;
  padding: ${props => props.$hasvalueinputconfirmemail === "true" ? "0px" : "13px"};
  padding-top: ${props => props.$hasvalueinputconfirmemail === "true" && "18px"};
  padding-left: ${props => props.$hasvalueinputconfirmemail === "true" && "17px"};
  padding-bottom: ${props => props.$hasvalueinputconfirmemail === "true" && "8px"};
  
`

export const ContainerEmailConfirmAndWarning = styled.div`
  width: 42%;
`
