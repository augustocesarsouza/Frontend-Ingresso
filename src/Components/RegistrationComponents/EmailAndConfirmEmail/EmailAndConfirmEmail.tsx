import * as Styled from './styled';
import { useState, useEffect, useRef } from 'react';
import InputWarning from '../InputWarning/InputWarning';
import EmailSvg from '../../../Svg/EmailSvg';
// import * as signalR from '@microsoft/signalr';

interface EmailAndConfirmEmailProps {
  verifyFields: boolean;
  setVerifyFields: React.Dispatch<React.SetStateAction<boolean>>;
  setValueOfEmail: React.Dispatch<React.SetStateAction<string>>;
  setValueOfEmailConfirm: React.Dispatch<React.SetStateAction<string>>;
}

const EmailAndConfirmEmail = ({
  verifyFields,
  setVerifyFields,
  setValueOfEmail,
  setValueOfEmailConfirm,
}: EmailAndConfirmEmailProps) => {
  const [FocusOrBlur, setFocusOrBlur] = useState(false);
  const [emailValid, setemailValid] = useState(true);

  const [hasValueInputName, setHasValueInputName] = useState(false);
  const [valueInputEmail, setValueInputEmail] = useState('');
  const [valueEmailUserConfirm, setValueEmailUserConfirm] = useState('');

  const [hasValueInputConfirmEmail, setHasValueInputConfirmEmail] = useState(false);
  const [FocusOrBlurEmailConfirm, setFocusOrBlurEmailConfirm] = useState(false);
  const [emailConfirm, setEmailConfirm] = useState(true);

  const refInputEmail = useRef<HTMLInputElement | null>(null);
  const refInputConfirmEmail = useRef<HTMLInputElement | null>(null);
  // const [connection, setConnection] = useState<signalR.HubConnection | null>(null);
  const [existEmail, setExistEmail] = useState(false);

  // const startConnection = async () => {
  //   const newConnection = new signalR.HubConnectionBuilder()
  //     .withUrl('https://localhost:7456/generalhub')
  //     .build();

  //   try {
  //     await newConnection.start();
  //     setConnection(newConnection);
  //   } catch (error) {
  //     console.error('Error establishing SignalR connection', error);
  //   }
  // };

  // useEffect(() => {
  //   if (connection === null) {
  //     startConnection();
  //   }
  // }, [connection]);

  // useEffect(() => {
  //   if (connection === null) return;
  //   if (valueInputEmail.includes('@gmail.com')) {
  //     connection.invoke('CheckEmailAlreadyExists', valueInputEmail);
  //   }

  //   connection.on('email-check-result', (exists: boolean) => {
  //     if (exists) {
  //       setExistEmail(true);
  //     } else {
  //       setExistEmail(false);
  //     }
  //   });
  // }, [connection, valueInputEmail]);

  useEffect(() => {
    if (verifyFields) {
      const valueEmail = refInputEmail.current.value;
      setValueOfEmail(valueEmail);
      if (!valueEmail.includes('@gmail.com')) {
        setVerifyFields(false);
        setemailValid(false);
      }

      const valueEmailConfirm = refInputConfirmEmail.current.value;
      setValueOfEmailConfirm(valueEmailConfirm);
      if (valueEmail.length <= 0) {
        setVerifyFields(false);
        setemailValid(false);
        setEmailConfirm(false);
      } else {
        if (!(valueEmail === valueEmailConfirm)) {
          setEmailConfirm(false);
          setemailValid(false);
        }
      }
    }
  }, [refInputConfirmEmail.current, refInputEmail.current, verifyFields]);

  const handleChangeInputName = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (e.target.value.length > 0) {
      setHasValueInputName(true);
      setValueInputEmail(e.target.value);
    } else {
      setHasValueInputName(false);
    }

    if (e.target.value.includes('@gmail.com')) {
      if (e.target.value.length >= 3) {
        setemailValid(true);
      } else {
        setemailValid(false);
      }
    } else {
      setemailValid(false);
    }

    if (valueEmailUserConfirm === e.target.value) {
      setHasValueInputConfirmEmail(true);
      setEmailConfirm(true);
    } else {
      setEmailConfirm(false);
    }
  };

  const handleChangeInputEmailConfirm = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValueEmailUserConfirm(e.target.value);
    if (e.target.value.length > 0) {
      setHasValueInputConfirmEmail(true);
    } else {
      setHasValueInputConfirmEmail(false);
    }

    if (e.target.value === valueInputEmail) {
      setHasValueInputConfirmEmail(true);
      setemailValid(true);

      if (e.target.value.length >= 3) {
        setEmailConfirm(true);
      } else {
        setEmailConfirm(false);
      }
    } else {
      setEmailConfirm(false);
      // setemailValid(false);
    }
  };

  const handleFocus = () => {
    setFocusOrBlur(true);
  };

  const handleBlur = () => {
    setFocusOrBlur(false);
  };

  const handleFocusEmailConfirm = () => {
    setFocusOrBlurEmailConfirm(true);
  };

  const handleBlurEmailConfirm = () => {
    setFocusOrBlurEmailConfirm(false);
  };

  return (
    <Styled.ContainerMainEmail>
      <Styled.ContainerWarning $container="1">
        <Styled.ContainerColorWarning
          $focusorblur={String(FocusOrBlur)}
          $emailvalid={String(emailValid)}
        >
          <Styled.ContainerEmailSvg $focus={String(FocusOrBlur)} $emailvalid={String(emailValid)}>
            <EmailSvg />
          </Styled.ContainerEmailSvg>
          <Styled.SpanWarning $hasvalueinput={String(hasValueInputName)}>
            *E-mail
          </Styled.SpanWarning>
          <Styled.Input
            type="text"
            id="input-email"
            ref={refInputEmail}
            onChange={handleChangeInputName}
            $hasvalueinputname={String(hasValueInputName)}
            onFocus={handleFocus}
            onBlur={handleBlur}
          />
        </Styled.ContainerColorWarning>
        <InputWarning
          Children={
            'Informe um endereço de e-mail válido. Ele será utilizado mais tarde para validar seus dados'
          }
          enableWarning={emailValid}
        />
      </Styled.ContainerWarning>
      <Styled.ContainerWarning $container="2">
        <Styled.ContainerColorWarning
          $focusorblur={String(FocusOrBlurEmailConfirm)}
          $emailvalid={String(emailConfirm)}
        >
          <Styled.ContainerEmailSvg
            $focus={String(FocusOrBlurEmailConfirm)}
            $emailvalid={String(emailConfirm)}
          >
            <EmailSvg />
          </Styled.ContainerEmailSvg>
          <Styled.SpanWarning $hasvalueinput={String(hasValueInputConfirmEmail)}>
            *Confirmação do E-mail
          </Styled.SpanWarning>
          <Styled.InputConfirmEmail
            type="text"
            id="input-confirm-email"
            ref={refInputConfirmEmail}
            onChange={handleChangeInputEmailConfirm}
            $hasvalueinputconfirmemail={String(hasValueInputConfirmEmail)}
            onFocus={handleFocusEmailConfirm}
            onBlur={handleBlurEmailConfirm}
          />
        </Styled.ContainerColorWarning>
        <InputWarning
          Children={'Este campo deve ser igual ao de e-mail'}
          enableWarning={emailConfirm}
        />
      </Styled.ContainerWarning>
    </Styled.ContainerMainEmail>
  );
};

export default EmailAndConfirmEmail;
