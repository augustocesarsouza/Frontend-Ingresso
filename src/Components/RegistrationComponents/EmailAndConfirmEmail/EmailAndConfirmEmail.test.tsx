import { render } from '@testing-library/react';
import EmailAndConfirmEmail from './EmailAndConfirmEmail';

describe('EmailAndConfirmEmail', () => {
  test('should render input email', () => {
    const { container } = render(<EmailAndConfirmEmail />);

    const spanElement = container.querySelector('#input-email');
    expect(spanElement).toBeInTheDocument();
  });
  EmailAndConfirmEmail;

  test('should render input confirm email', () => {
    const { container } = render(<EmailAndConfirmEmail />);

    const spanElement = container.querySelector('#input-confirm-email');
    expect(spanElement).toBeInTheDocument();
  });

  test('Snapshot', () => {
    const { container } = render(<EmailAndConfirmEmail />);

    expect(container).toMatchSnapshot();
  });
});
