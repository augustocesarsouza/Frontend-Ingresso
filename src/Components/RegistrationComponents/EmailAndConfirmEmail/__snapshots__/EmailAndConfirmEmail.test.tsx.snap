// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailAndConfirmEmail Snapshot 1`] = `
.c0 {
  display: flex;
  gap: 20px;
}

.c1 {
  width: 55%;
}

.c6 {
  width: 42%;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  border-width: 1px 1px 1px 5px;
  border-style: solid;
  border-color: #8080804d;
  border-color: #8080804d;
  margin-bottom: 6px;
}

.c3 {
  left: 0;
  top: 0;
  width: 45px;
  height: 22px;
  border-right: 1px solid rgb(221, 221, 221);
}

.c3 svg {
  width: 100%;
  height: 100%;
  fill: #8080804d;
}

.c4 {
  position: absolute;
  left: 53px;
  top: 13px;
  font-size: 13px;
  color: #a4a2a2;
  pointer-events: none;
  transition: top 0.2s ease,font-size 0.2s ease;
}

.c5 {
  border: none;
  width: 100%;
  height: 100%;
  outline: none;
  padding: 13px;
}

.c7 {
  border: none;
  width: 100%;
  height: 100%;
  outline: none;
  padding: 13px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            class="styled__CredentialIcon-btCXJl kuwuSY"
          >
            <svg
              role="img"
              viewBox="0 0 32 33"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15.403 19.613-2.127-2a.844.844 0 0 1-.065.065l-5.37 4.836H24.16l-5.37-4.836a.844.844 0 0 1-.065-.065l-2.127 2a.875.875 0 0 1-1.194 0zM16 17.88l7.29-6.852H8.71L16 17.881zm8.519-5.713L19.94 16.47a.858.858 0 0 1 .012.01l4.566 4.112v-8.424zM7.48 20.592l4.566-4.112a.858.858 0 0 1 .012-.01L7.48 12.168v8.424zm-1.703-9.303c0-1.05.789-1.9 1.76-1.9H24.46c.973 0 1.761.858 1.761 1.9v10.965c0 1.05-.789 1.9-1.76 1.9H7.538c-.973 0-1.761-.857-1.761-1.9V11.29z"
                stroke-width="0.5"
              />
            </svg>
          </svg>
        </div>
        <span
          class="c4"
        >
          *E-mail
        </span>
        <input
          class="c5"
          id="input-email"
          type="text"
        />
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            class="styled__CredentialIcon-btCXJl kuwuSY"
          >
            <svg
              role="img"
              viewBox="0 0 32 33"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15.403 19.613-2.127-2a.844.844 0 0 1-.065.065l-5.37 4.836H24.16l-5.37-4.836a.844.844 0 0 1-.065-.065l-2.127 2a.875.875 0 0 1-1.194 0zM16 17.88l7.29-6.852H8.71L16 17.881zm8.519-5.713L19.94 16.47a.858.858 0 0 1 .012.01l4.566 4.112v-8.424zM7.48 20.592l4.566-4.112a.858.858 0 0 1 .012-.01L7.48 12.168v8.424zm-1.703-9.303c0-1.05.789-1.9 1.76-1.9H24.46c.973 0 1.761.858 1.761 1.9v10.965c0 1.05-.789 1.9-1.76 1.9H7.538c-.973 0-1.761-.857-1.761-1.9V11.29z"
                stroke-width="0.5"
              />
            </svg>
          </svg>
        </div>
        <span
          class="c4"
        >
          *Confirmação do E-mail
        </span>
        <input
          class="c7"
          id="input-confirm-email"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
